api/services/StatisticsMassifService.js

```diff
@@ -0,0 +1,231 @@
+const CREATE_MATERIALIZED_VIEW = `
+  CREATE MATERIALIZED VIEW v_massif_info AS
+  SELECT m.id as id_massif,
+  c.id as id_cave,
+  n.name as name_cave,
+  c.depth as depth_cave,
+  c.length as length_cave,
+  c.is_diving as is_diving_cave,
+  COUNT(DISTINCT e.id) as nb_entrances
+  FROM t_massif m 
+  JOIN t_entrance e ON ST_Contains(ST_SetSRID(m.geog_polygon::geometry, 4326), ST_SetSRID(ST_MakePoint(e.longitude, e.latitude), 4326))
+  JOIN t_cave c ON e.id_cave = c.id 
+  JOIN t_name n ON n.id_cave = c.id 
+  WHERE n.is_main = true 
+  AND c.is_deleted = false
+  AND e.is_deleted = false
+  AND m.is_deleted = false
+  GROUP BY(m.id, c.id, n.name, c.depth, c.length, c.is_diving );
+`;
+
+const GET_NB_CAVES = `
+  SELECT COUNT(*) as nb_caves
+  FROM v_massif_info
+  WHERE id_massif = $1
+`;
+
+const GET_NB_NETWORKS = `
+  SELECT COUNT(*) as nb_networks
+  FROM v_massif_info
+  WHERE id_massif = $1
+  AND nb_entrances > 1
+`;
+
+const FIND_CAVE_WITH_MAX_DEPTH_IN_MASSIF = `
+  SELECT id_cave, name_cave, depth_cave
```

Comment:
Just a comment (your option is maybe better than mine ;) ). COuld it be better to do something like this (untested) : 

SELECT id_cave, name_cave, depth_cave
  FROM v_massif_info
  WHERE id_massif = $1
ORDER BY depth_cave desc
LIMIT 1;

Reply:
We thought initially that using the MAX function was clearer. 
We tested both our option and yours, it doesn't make a huge difference, yours is at 0.3s and ours 0.28s, since we don't have a large amount of data. We looked on the net and some ppl found that the `MAX ` is optimized for this kind of request and that on some cases, the `ORDER BY` is asking more ressources.